services/movie.py

```diff
@@ -1,21 +1,37 @@
+from typing import List, Optional
+
 from django.db.models import QuerySet
 
-from db.models import Movie
+from db.models import Movie, Genre, Actor
+
+
+def get_movies(title: Optional[str] = None,
+               genres_ids: Optional[List[int]] = None,
+               actors_ids: Optional[List[int]] = None) -> QuerySet:
+
+    """
+    Fetches a list of movies based on the given filters.
+
+    Args:
+        title (str): The title of the movie to search for.
+        genres_ids (list): A list of genre IDs to filter by.
+        actors_ids (list): A list of actor IDs to filter by.
 
+    Returns:
+        QuerySet: A Django QuerySet of filtered movies.
+    """
+    movies = Movie.objects.all()
 
-def get_movies(
-    genres_ids: list[int] = None,
-    actors_ids: list[int] = None,
-) -> QuerySet:
-    queryset = Movie.objects.all()
+    if title:
+        movies = movies.filter(title__icontains=title)
 
     if genres_ids:
-        queryset = queryset.filter(genres__id__in=genres_ids)
+        movies = movies.filter(genres__id__in=genres_ids).distinct()
```

Comment:
Ensure that the `Movie` model has a `genres` field if you intend to filter by genres. If the `genres` field is not defined, this line will cause an error.